### ISSUE TITLE
Modifying profiles.json causes multiple settings reloads

### ISSUE THREAD

**Issue by d-bingham** [microsoft/terminal#858]
This bug-tracker is monitored by Windows Console development team and other technical types.  **We like detail!**

If you have a feature request, please post to [the UserVoice](https://wpdev.uservoice.com/forums/266908).  

> **Important: When reporting BSODs or security issues, DO NOT attach memory dumps, logs, or traces to Github issues**. Instead, send dumps/traces to secure@microsoft.com, referencing this GitHub issue.

Please use this form and describe your issue, concisely but precisely, with as much detail as possible

* Your Windows build number:  Microsoft Windows [Version 10.0.18362.86]

* What you're doing and what's happening: When editing profiles.json, saving a _modified_ version of the file generates multiple app-wide settings refreshes.

* What's wrong / what should be happening instead: Modifying the settings file should only generate one app-wide settings update

**Comment by fearthecowboy** [comment]
It looks like in `TerminalApp::CascadiaSettings`  it has 

``` c++
  static std::unique_ptr<CascadiaSettings> LoadAll(const bool saveOnLoad = true);
```

`saveOnLoad` is `true` and it's only ever called with the default value. 

I'd think that this should not save by default - if it loaded, it should be fine to load later, there's little gained by saving it (which also reformats the json. GRRRRR 😠 ).

**Comment by zadjii-msft** [comment]
Yea I think I actually have a fix for this in #748 [here](https://github.com/microsoft/terminal/pull/748/files#diff-a011f61f234a66ac8dd94efe8e488244R408)

**Comment by d-bingham** [comment]
Yeah, the fix in #748 fixes the problem.
